[ci] Enable LTO in release builds

metajack opened the pull request:
LTO was disabled by default in #4829. This PR overrides the LTO settings in CI for the release profile. This will keep local developers building quickly without LTO while still testing real release conditions in CI.

Fixes #4987

andll commented:
One note - we were using thin lto(lto=thin, instead of true) previously, this seemed like a good choice between TPS and compile speed, although I don't have exact numbers

thefallentree commented:
> One note - we were using thin lto(lto=thin, instead of true) previously, this seemed like a good choice between TPS and compile speed, although I don't have exact numbers

thin was chosen mainly to enable fast development cycles, but for CI it doesn't matter , and fat LTO gives more performance back.

thefallentree commented:
the error looks like an OOM issue.

andll commented:
To the best of my knowledge, this PR does not affect performance of prod at all, this is just about setting up LTO in unit tests

It is also not true that 'performance does not matter for CI', because CI blocks diffs landing. We do consider land time to be important, for example in cluster test land blocking, I don't understand why do we have different view here

metajack commented:
In the interest of streamlining CI, I moved the flag to build-common.sh which will enable LTO for all release binaries, and removed all our CI workflow that builds unused release stuff. LBT will still catch build failures under --release with LTO on.

metajack commented:
@bors-libra r=rexhoffman

bors-libra commented:
:pushpin: Commit eacd11b has been approved by `rexhoffman`

<!-- @bors-libra r=rexhoffman eacd11b2ea4345393702e75c71750fde76f753e3 -->

bors-libra commented:
:hourglass: Testing commit eacd11b2ea4345393702e75c71750fde76f753e3 with merge 29aafa83d5bb5c17525f2fb233af5ef19e282896...

bors-libra commented:
:sunny: Test successful - [checks-actions_land_blocking_test](https://github.com/libra/libra/runs/899540856), [checks-circle_commit_workflow](https://circleci.com/workflow-run/0fb58092-9a31-493a-9fd0-054d9f19848e?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)
Approved by: rexhoffman
Pushing 29aafa83d5bb5c17525f2fb233af5ef19e282896 to master...

metajack commented:
20% to TPS it seems. Yay :)

cc @andll
